timur/lib/client/jsx/components/timur_nav.jsx

```diff
@@ -2,120 +2,64 @@
-      >
-        <div className="image"></div>
-        <Halo radius={27.5} />
-      </div>
-    </a>
+const Logo = connect(({exchanges}) => ({exchanges}))(({exchanges}) =>
+  <div id='timur-logo'
+    className={Object.keys(exchanges).length > 0 ? 'throb' : null }>
+    <div className="image"/>
+    <Halo radius={25} />
   </div>
 );
 
-const ICONS = {
-  administrator: 'user-astronaut',
-  editor: 'user-edit',
-  viewer: 'user'
-};
-
-const Login = ({user}) => {
-  let {first, last, permissions} = user;
-
-  let project_perm = permissions[CONFIG.project_name];
-  let role = project_perm ? project_perm.role : '';
-  if (!user) return null;
-  return (
-    <div id="login">
-      {first} {last}
-      <i className={`fa fa-${ICONS[role]}`} title={role} />
-    </div>
-  );
-};
-
-class TimurNav extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-  }
-
-  renderTabs() {
-    let {mode} = this.props;
-    let tabs = {
-      browse: Routes.browse_path(CONFIG.project_name),
-      search: Routes.search_path(CONFIG.project_name),
-      map: Routes.map_path(CONFIG.project_name),
-      manifests: Routes.manifests_path(CONFIG.project_name),
-      plots: Routes.plots_path(CONFIG.project_name),
-      help: 'https://mountetna.github.io/timur.html'
-    };
-
-    return Object.keys(tabs).map((tab_name) => (
-      <div
-        key={tab_name}
-        className={`nav_tab ${mode == tab_name ? 'selected' : ''}`}
-      >
-        <Link link={tabs[tab_name]}>{tab_name}</Link>
-      </div>
-    ));
-  }
+const getTabs = () => ({
+  browse: Routes.browse_path(CONFIG.project_name),
+  search: Routes.search_path(CONFIG.project_name),
+  map: Routes.map_path(CONFIG.project_name),
+  manifests: Routes.manifests_path(CONFIG.project_name),
+  plots: Routes.plots_path(CONFIG.project_name),
+  help: 'https://mountetna.github.io/timur.html'
+});
 
-  render() {
-    let {exchanges, mode, environment, user} = this.props;
-    let isDev = false; //environment == 'development';
-    let heading = (
-      <span>
-        {'Timur'}
-        <b>{' : '}</b>
-        {isDev ? 'Development' : 'Data Browser'}
-      </span>
-    );
-    let logo_id = isDev ? 'dev' : 'normal';
+const ModeBar = ({mode}) =>
+  <div id='nav'>
+    {
+      Object.entries(getTabs()).map(([tab_name, route]) =>
+        <div key={tab_name}
+          className={`nav_tab ${mode == tab_name ? 'selected' : ''}`}>
+          <Link link={route}>{tab_name}</Link>
+        </div>
+      )
+    }
+  </div>;
 
-    return (
-      <div id="header">
-        <Logo exchanges={exchanges} logo_id={logo_id} />
-        {mode !== 'home' && <IdentifierSearch />}
-        <div id="nav">{mode !== 'home' && this.renderTabs()}</div>
-        <Login user={user} />
-      </div>
-    );
-  }
-}
+const TimurNav = ({mode, user}) =>
+  <Nav user={user} logo={Logo} app='timur'>
+    {mode !== 'home' && <IdentifierSearch />}
```

Comment:
Just more like an FYI, but you can use `<React.Fragment>` here to compose siblings:

```
{mode !== 'home' && <React.Fragment>
   <IdentifierSearch />
   <ModeBar mode={mode}/>
</React.Fragment>}
```

This is another one of those super minor things that isn't a big deal either way, but just interesting to share as it can be more useful than a continuous chain of conditions in some cases.

Reply:
I've seen React.Fragment in places but haven't really used it yet. Thanks for the tip!